src/block.h

```diff
@@ -5,46 +5,166 @@
 #include "db.h"
 
 /// [issue](https://github.com/WSU-CEG-6110-4410/Remainders-Craft/issues/8)
-/// These functions were derived from main.c. Further documentation is necessary.
-
+/// [issue](https://github.com/WSU-CEG-6110-4410/Remainders-Craft/issues/21)
+
+/// Use this function to remove a sign face from a chunk.
+///\param[in] x: The x coordinate of the sign.
+///\param[in] y: The y coordinate of the sign.
+///\param[in] z: The z coordinate of the sign.
+///\param[in] face: The face of the sign (0,1,2,3,4).
+///\param[in,out] model: The current game instance to be modified.
 void unset_sign_face(int x, int y, int z, int face, Model *model);
 
+/// Use this function to set the sign text of a chunk.
+///\param[in] p: The x coordinate for the chunk.
+///\param[in] q: The y coordinate for the chunk.
+///\param[in] x: The x coordinate of the sign.
+///\param[in] y: The y coordinate of the sign.
+///\param[in] z: The z coordinate of the sign.
+///\param[in] face: The face of the sign (0,1,2,3,4).
+///\param[in] text: The text content to be added.
+///\param[in] dirty: 1 or 0. Indicates whether the chunk should
```

Comment:
The dirty bit here might be to tell the game that something has changed in the chunk, but I'm not too sure. I know before it was used to keep something rendered, but this might be a check for changes being made to the chunk as well. Some of the functions look to call the dirty_chunk function, but ones where it gets passed in seem to just set a chunks dirty value.

Reply:
Yeah, it seems strange that sometimes it calls dirty_chunk() but other times it sets dirty manually. My only guess is that dirty_chunk() can dirty the surrounding chunks, but setting dirty directly only affects a single chunk. So for setting a sign maybe it's ensuring that only the sign chunk is dirtied. I'm not sure why that's necessary though. I'm inclined to keep it as is since that's what we documented dirty to mean elsewhere.

Reply:
Yeah that makes sense. I remember the actual function doing more than just setting a single value. Sounds right for it to want to keep track of the single chunk being changed due to the sign. I say the documentation is good then.